app/javascript/multi-stamp/components/MultiStampLibrarySplitter.js

```diff
@@ -0,0 +1,119 @@
+import MultiStamp from './MultiStamp'
+import Vue from 'vue'
+import { 
+  checkSize, checkDuplicates, checkMaxCountRequests, 
+  checkLibraryTypesInAllWells,
+  checkAllRequestsWithSameReadySubmissions,
+  checkPlateWithSameReadySubmissions,
+  checkMinCountRequests, checkAllSamplesInColumnsList/* checkExcess */ 
+} from 'shared/components/plateScanValidators'
+import { baseTransferCreator } from 'shared/transfersCreators'
+
+export default Vue.extend({
+  mixins: [MultiStamp],
+  props: {
+    childrenLibraryTypeToPurposeMappingJson: {
+      default: '{}',
+      validator(value) {
+        try {
+          JSON.parse(value)
+        } catch (e) {
+          return false
+        }
+        return true
+      }
+    }
+  },
+  computed: {
+    valid() {
+      return this.unsuitablePlates.length === 0 // None of the plates are invalid
+             && this.validTransfers.length > 0 // We have at least one transfer
+             && this.excessTransfers.length === 0 // No excess transfers
+             && this.duplicatedTransfers.length === 0 // No duplicated transfers
+             && (this.transfersCreatorObj.isValid)
+    },
+    childrenLibraryTypeToPurposeMapping() {
+      return JSON.parse(this.childrenLibraryTypeToPurposeMappingJson)
+    },
+    libraryTypes() {
+      return Object.keys(this.childrenLibraryTypeToPurposeMapping)
+    },
+    validTransfersByTargetPlate() {
+      return this.validTransfers.reduce((memo, transfer) => { 
+        if (typeof memo[transfer.targetPlate] === 'undefined') {
+          memo[transfer.targetPlate] = []
+        }
```

Comment:
I didn't know about this operator. Seems a good way or reducing all that typeof code. I'll add it

Reply:
It's quite a new addition, had to double check the Babel support. 

I always ended up assuming it existed (well ||= at least) because it does in Ruby, and then fall foul when it didn't work. Was very glad when it was added.